src/terrain/terrain_base.hpp

```diff
@@ -257,7 +375,10 @@ class TerrainBase : public voxel_utility::VoxelBase {
      */
     inline bool
     in_range(int x, int y, int z) const {
-        return (x < X_MAX && x >= 0 && y < Y_MAX && y >= 0 && z < Z_MAX && z >= 0);
+        return (
+            (uint32_t)x < X_MAX && x >= 0 && (uint32_t)y < Y_MAX && y >= 0
+            && (uint32_t)z < Z_MAX && z >= 0
```

Comment:
Wrong cast
Also you cast inconsistantly (for the lt but not the gt)
Also from our discussion in discord these are uint16_t -> int -> uint32_t, which can cause problems with negative numbers